01_loaddata.R

```diff
@@ -12,80 +12,109 @@ library(lubridate)
 ## DATA PRE-PROCESSING ##
 # Read CSV Data
 
-
-# TODO: add a purrr::map to be able to loop through many files/birds, 
-# need to add a value to distinguish the tag id.
-
-
-
-birds <- read_csv(path("01_inputs", "DS 2 _P7165_Con_213835_250122134729.csv"), name_repair = "unique", locale = locale(encoding = "latin1"))
-
-# Convert to SF Object
-birds_sf <- st_as_sf(birds, coords = c("Longitude", "Latitude"), crs = 4326, remove = FALSE)
-
-# Filter Out Bad Data and Sort by Date Ascending
-birds_sf <- birds_sf %>% filter(CRC != "Fail")
-
-birds_sf <- birds_sf |>
-  dplyr::mutate(datetime = dmy_hms(paste(Date, Time, sep = " ")))
-
-birds_sf <- birds_sf %>% arrange(datetime)
-
-birds_sf <- birds_sf |>
-  mutate(
-    location.long_prior = lag(Longitude, 1L),
-    location.lat_prior = lag(Latitude, 1L)
-  )
-
-
-# Calculate Distance Between Sequential Ping Locations
-birds_dist <- st_distance(birds_sf) # Creates a distance matrix for every point's distance to every other point in the dataset (in metres)
-
-dist_seq <- c() # creates an empty vector to store sequential distances
-
-for (i in 1:nrow(birds_sf)) { # loops through every GPS ping in the dataset
-  if (i == 1) { # if it's the very first point in the dataset, it will calculate the very first distance calculation, which will be zero
-    dist_seq[i] <- birds_dist[i, i]
-  } else { # any other element will be calculating the distance between the current point and the previous ping location
-    dist_seq[i] <- birds_dist[i, i - 1]
+### List all bird files in the inputs folder
+bird_files <- list.files(path("01_inputs/bird_data"), recursive = TRUE, pattern = ".csv") # Read in a list of all bird CSV data
+
+bird_files <- bird_files[1:60]
+
+### Loop through each existing bird CSV file and perform the analyses
+birds_processed <- purrr::map(bird_files, function(b) {
+  # testing
+  # b <- bird_files[1]
+  
+  bname <- gsub(".*/(.*)", "\\1", b)
+  
+  cli::cli_alert_info("processing {bname}")
+  
+  ## 0) Data Pre-Processing
+  # Read in CSV Data
+  birds <- read_csv(path("01_inputs/bird_data", b), name_repair = "unique", locale = locale(encoding = "latin1"))
+  
+  # Filter Out Bad Data and Sort by Date Ascending
+  birds <- birds %>% filter(CRC != "Fail")
+  
+  ## Use an if statement to skip records that no longer have at least 3 records after filtering bad data
+  if(nrow(birds) >= 3){
```

Comment:
I like this option, although we may want to generalize to make sure we capture other cases. Lets run through the files and see how what errors come up!